orbit/pkg/table/user_exec/exec.go

```diff
@@ -0,0 +1,144 @@
+// build +darwin
+
+package user_exec
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"os"
+	"os/exec"
+	"os/user"
+	"time"
+
+	"github.com/kolide/launcher/pkg/osquery/tables/tablehelpers"
+	"github.com/osquery/osquery-go/plugin/table"
+	"github.com/rs/zerolog/log"
+)
+
+// ExecOsqueryLaunchctl runs osquery under launchctl, in a user context.
+func ExecOsqueryLaunchctl(ctx context.Context, timeoutSeconds int, username string, osqueryPath string, query string) ([]byte, error) {
+	ctx, cancel := context.WithTimeout(ctx, time.Duration(timeoutSeconds)*time.Second)
+	defer cancel()
+
+	targetUser, err := user.Lookup(username)
+	if err != nil {
+		return nil, fmt.Errorf("looking up username %s: %w", username, err)
+	}
+
+	cmd := exec.CommandContext(ctx,
+		"launchctl",
+		"asuser",
+		targetUser.Uid,
+		osqueryPath,
+		"--config_path", "/dev/null",
+		"--disable_events",
+		"--disable_database",
+		"--disable_audit",
+		"--ephemeral",
+		"-S",
+		"--json",
+		query,
+	)
+
+	dir, err := os.MkdirTemp("", "osq-launchctl")
+	if err != nil {
+		return nil, fmt.Errorf("mktemp: %w", err)
+	}
+	defer os.RemoveAll(dir)
+
+	if err := os.Chmod(dir, 0o755); err != nil {
```

Comment:
How about change the owner ([os.Chown](https://pkg.go.dev/os#Chown)) , instead of adding read and execution permissions to `Other`s?

Reply:
`os.Chown` requires the uid and gid of the user, which requires more effort to get. I also can't
find a good way to set ACLs on macOS without an extra library.

Reply:
I think we need to be really careful about this. Is it possible with `0755` permissions that another user could modify the executable before it's executed resulting in privilege escalation?

Reply:
The executable is not copied to the directory, it is only used as the current working directory. I'm
having trouble finding out exactly what it is used for, I'll do some experimentation around not
making a temp dir.